src/main/java/org/folio/qm/messaging/listener/DataImportEventListener.java

```diff
@@ -36,6 +42,8 @@ public void diCompletedListener(DataImportEventPayload data, MessageHeaders mess
     concurrency = "#{folioKafkaProperties.listener['di-error'].concurrency}",
     containerFactory = "dataImportKafkaListenerContainerFactory")
   public void diErrorListener(DataImportEventPayload data, MessageHeaders messageHeaders) {
-    processingService.processDIError(data);
+    if(!deleteEventProcessingService.processDIError(data)){
+      importEventProcessingService.processDIError(data);
+    }
```

Comment:
Not very clean when we call each service.
If it's not possible to delete record, we do import?
Lets do something like this:
```suggestion
    DeferredResult deferredResult = deleteCache.getFromCache(jobExecutionId);
    if (deferredResult != null) {
      deleteEventProcessingService.processDICompleted(data)
    } else {
      importEventProcessingService.processDICompleted(data);
    }
```